pyshelf/search/update_manager.py

```diff
@@ -29,6 +29,41 @@ def remove_unlisted_documents(self, ex_key_list):
 
         return self._bulk_delete(query)
 
+    def remove_unlisted_documents_wildcard(self, key, value_list):
+        """
+            Removes all documents that do not match the query built.
+
+            Args:
+                key(string): key of metadata property to use to query.
+                value_list(list[string]): list of values to match against
+
+            Returns:
+                int: number of records that were deleted
+
+            Example:
+                remove_unlisted_documents_wildcard("artifactPath", ["/test/artifact/*"])
+                In this example all documents that don't match artifactPath=/test/artifact/*
+                are pruned.
+        """
+        query = None
+
+        for value in value_list:
+            nested_query = Q(SearchType.MATCH, property_list__name=key)
+            nested_query &= Q(SearchType.WILDCARD, property_list__value=value)
+
+            # Initialize the query.... You would  think you can init query = ~Q() and then use
+            # the |= operator but that does not work. You have to wrap the query at the end.
+            if not query:
```

Comment:
Instead of doing this, can you do something like

```
additional_query = Q("nested", path="property_list", query=nested_query)
if not query:
    query = additional_query
else:
    query |= additional_query